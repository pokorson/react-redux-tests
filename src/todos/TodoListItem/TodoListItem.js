import React from 'react';

class TodoListItem extends React.PureComponent {
  handleRemove = () => {
    this.props.onRemove(this.props.id);
  };

  handleComplete = () => {
    this.props.onComplete(this.props.id);
  };

  render() {
    return (
      <div>
        <span>{this.props.title}</span>
        <input type='checkbox' value={this.props.checked} onChange={this.handleComplete} />
        <button onClick={this.handleRemove}>remove</button>
      </div>
    );
  }
}

export default TodoListItem;
