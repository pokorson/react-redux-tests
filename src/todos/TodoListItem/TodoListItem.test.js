import React from 'react';
import TodoListItem from './TodoListItem';
import { shallow, render } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('TodoListItem component', () => {
  it('renders title from props', () => {
    const component = render(<TodoListItem title='test title' />);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('renders checked when props.checked is false', () => {
    const component = render(<TodoListItem checked={false} />);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('renders checked when props.checked is true', () => {
    const component = render(<TodoListItem checked={true} />);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('calls onComplete prop on checkbox change', () => {
    const props = {
      id: 1,
      onComplete: jest.fn(),
    };
    const component = shallow(<TodoListItem {...props} />);
    component.find('input').simulate('change');
    expect(props.onComplete).toHaveBeenCalledWith(props.id);
  });

  it('calls onRemove prop on button click', () => {
    const props = {
      id: 1,
      onRemove: jest.fn(),
    };
    const component = shallow(<TodoListItem {...props} />);
    component.find('button').simulate('click');
    expect(props.onRemove).toHaveBeenCalledWith(props.id);
  });
});
