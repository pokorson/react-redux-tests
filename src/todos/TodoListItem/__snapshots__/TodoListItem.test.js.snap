// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoListItem component renders checked when props.checked is false 1`] = `
<div>
  <span />
  <input
    type="checkbox"
    value="false"
  />
  <button>
    remove
  </button>
</div>
`;

exports[`TodoListItem component renders checked when props.checked is true 1`] = `
<div>
  <span />
  <input
    type="checkbox"
    value="true"
  />
  <button>
    remove
  </button>
</div>
`;

exports[`TodoListItem component renders title from props 1`] = `
<div>
  <span>
    test title
  </span>
  <input
    type="checkbox"
  />
  <button>
    remove
  </button>
</div>
`;
